onnxruntime/core/providers/mkldnn/subgraph/mkldnn_sum.h

```diff
@@ -0,0 +1,172 @@
+// Copyright(C) 2019 Intel Corporation
+// Licensed under the MIT License
+
+#pragma once
+#include "core/framework/op_kernel.h"
+#include "core/providers/mkldnn/mkldnn_fwd.h"
+#include "core/providers/mkldnn/mkldnn_common.h"
+#include "core/providers/mkldnn/subgraph/mkldnn_kernel.h"
+#include "core/util/math.h"
+
+namespace onnxruntime {
+namespace mkl_dnn {
+
+template <typename T>
+class MklDnnSum : public MklDnnKernel {
+ public:
+  explicit MklDnnSum(MklDnnNode& node,
+                  MKLDNNExecutionProvider* provider,
+                     const NodeAttributes& attributes,
+                     const std::string attributes_prefix = "") : MklDnnKernel(node, provider) {
+    ReadAttributes(attributes, attributes_prefix);
+  }
+
+  Status CreatePrimitives(const OrtCustomOpApi* api,
+                          OrtKernelContext* context,
+                          mkldnn::engine& cpu_engine,
+                          std::vector<mkldnn::primitive>& net,
+                          mkldnn::memory::format& source_format) override {
+    Ort::CustomOpApi ort{*api};
+    int num_inputs = mklnode_ptr_->num_inputs;
+    int input_index = mklnode_ptr_->input_start_index < 0 ? 0 : mklnode_ptr_->input_start_index;
+
+    std::vector<float> coeff;
+    TensorShape x_shape;
+    if (mklnode_ptr_->parent_nodes.size() == 0) {
+      const OrtValue* input_tensor = ort.KernelContext_GetInput(context, input_index);
+      auto tensor_info = ort.GetTensorTypeAndShape(input_tensor);
+      auto tensor_shape = ort.GetTensorShape(tensor_info);
+      ort.ReleaseTensorTypeAndShapeInfo(tensor_info);
+      auto xshape = tensor_shape.data();
+      auto xdim = tensor_shape.size();
+
+      ort_source_format_ = GetSourceFormat(static_cast<int>(xdim));
+      source_format = ort_source_format_;
+      src_format_ = ort_source_format_;
+      x_shape = TensorShape(xshape, xdim);
+    } else {
+      x_shape = parents_[0].get()->primitive_dst_shape_;
+      src_format_ = parents_[0].get()->primitive_dst_format_;
+    }
+    primitive_dst_shape_ = TensorShape(x_shape);
+
+    mkldnn::memory::dims dst_dims_mkl(
+        primitive_dst_shape_.GetDims().begin(), primitive_dst_shape_.GetDims().end());
+
+    for (int i = 0; i < num_inputs; i++) {
+      TensorShape x_shape1;
+
+      if (mklnode_ptr_->parent_nodes.size() == 0) {
+        const OrtValue* input_tensor = ort.KernelContext_GetInput(context, input_index);
+        auto tensor_info = ort.GetTensorTypeAndShape(input_tensor);
+        auto tensor_shape = ort.GetTensorShape(tensor_info);
+        ort.ReleaseTensorTypeAndShapeInfo(tensor_info);
+        auto xshape = tensor_shape.data();
+        auto xdim = tensor_shape.size();
+        mkldnn::memory::dims dims(xdim);
+
```

Comment:
formatting seems strange here?
are you using clang formatting to adhere to https://github.com/microsoft/onnxruntime/blob/master/docs/Coding_Conventions_and_Standards.md ?

Reply:
Yes. I am using clang with VS. I formatted code again.